tool/mapeditor.re

```diff
@@ -25,19 +39,21 @@ static MAP_EDITOR_COMPONENT = Component {
     on_element_pressed: fn(index: ElementIndex) {},
     on_element_released: fn(index: ElementIndex) {},
     on_key_down: fn(key: KeyCode, is_repeat: bool) {
-        if key.to_small() == KEY_TAB.to_small() {
-            enter_ui(create_map_editor_input_ui());
-        }
-        if key.to_small() == KEY_E.to_small() {
-            let index = match Tas::get_looked_at_element_index() {
-                Option::Some(index) => index,
-                Option::None => return,
-            };
-            let element = match try_get_element(index) {
-                Result::Ok(element) => element,
-                _ => return,
-            };
-            enter_ui(create_map_editor_element_ui(element, index, 0));
+        if MAP_EDITOR_STATE.mode == MapMode::Edit {
```

Comment:
Change this if to a guard-if. Reduces indentation by 1 and shows the reader that they don't need to wait for a possible `else` branch or other behaviour on the contrary condition.